kubernetes/kubernetes.go

```diff
@@ -608,12 +636,27 @@ func ensureMetricServicesAvailable(config KubeAgentConfig) (KubeAgentConfig, err
 	}
 
 	if !config.RetrieveNodeSummaries && !config.CollectHeapsterExport {
+		log.Debugf("Unable to retrieve data due to metrics-agent configuration.\n" +
+			"May be caused by cluster security mis-configurations or the RBAC role in the Cloudability " +
+			"namespace needs to be updated.\n" +
+			"Please confirm with your cluster security administrators that the " +
+			"RBAC role is able to work within your cluster's security configurations.")
 		return config, fmt.Errorf("unable to retrieve node summaries or heapster export: %s", err)
 	}
 
 	return config, nil
 }
 
+func handleNodeSourceError(err error) string {
+	var error string
+	if strings.Contains(err.Error(), "Please verify RBAC roles") {
```

Comment:
I like these options to add additional info to the logs where we know more about the problem 👍